submissions/SerhiiYakovenko/js-dom/script.js

```diff
@@ -0,0 +1,128 @@
+        You can customize your character’s cyberware, skillset and playstyle, and explore a vast city where the choices you make shape the story and the world around you.`,
+  },
+  {
+    heading: "Mercenary",
+    img: "img/mercenary.jpg",
+    text:
+      "Become a cyberpunk, an urban mercenary equipped with cybernetic enhancements and build your legend on the streets of Night City.",
+  },
+  {
+    heading: "Implants",
+    img: "img/implant.jpg",
+    text:
+      "Take the riskiest job of your life and go after a prototype implant that is the key to immortality.",
+  },
+];
+
+const body = document.querySelector("body");
+
+// Header
+const header = document.createElement("header");
+let h1 = document.createElement("h1");
+h1.textContent =
+  "Welcome to the CyberDOM. Here you can get more info about Cyberpunk 2077 game.";
+header.appendChild(h1);
+body.appendChild(header);
+
+// Nav and Main
+const nav = document.createElement("nav");
+nav.classList.add("nav");
+const navList = document.createElement("ul");
+navList.classList.add("nav__items");
+nav.appendChild(navList);
+
+const main = document.createElement("main");
+main.classList.add("main");
+
+// Items
+items.forEach((item) => {
+  const navItem = createElementWithClassName("li", item.heading, "nav__item");
+  const mainItem = createItem(item);
+  navList.appendChild(navItem);
+  main.appendChild(mainItem);
+});
+
+body.appendChild(nav);
+body.appendChild(main);
+
+// Footer
+const footer = document.createElement("footer");
+let p = document.createElement("p");
+p.insertAdjacentHTML(
+  "afterbegin",
+  `Created by <a href="https://github.com/serhiiyakovenko" target="_blank">serhiiyakovenko</a> for <a href="https://kottans.org/" target="_blank">kottans</a>`
+);
+footer.appendChild(p);
+body.appendChild(footer);
+
+//Functions
+function createElementWithClassName(tag, value, className) {
+  const element = document.createElement(tag);
+  element.classList.add(className);
+  switch (tag) {
+    case "img":
+      element.src = value;
+      element.alt = value.replace("img/", "").replace(".jpg", "");
+      break;
+    default:
+      element.textContent = value;
+      break;
+  }
+  return element;
+}
+
+function createItem(item) {
+  const div = document.createElement("div");
+  div.classList.add(
+    "main__item",
+    item.heading.toLowerCase().split(" ").join("_")
+  );
+  const mainItemContent = createElementWithClassName(
+    "p",
+    item.text,
+    "main__content"
+  );
+  div.appendChild(mainItemContent);
+  const mainItemImage = createElementWithClassName(
+    "img",
+    item.img,
+    "main__image"
+  );
+  div.appendChild(mainItemImage);
+  div.classList.add("hidden");
+  return div;
+}
+
+// Hide and unhide items
+const navSection = document.getElementsByClassName("nav__item");
+for (var i = 0; i < navSection.length; i++) {
+  navSection[i].addEventListener("click", function () {
```

Comment:
use [event delegation](https://javascript.info/event-delegation), please.
It will help you to increase the performance of your app and use only one event listener instead of the `navSection.length`

Reply:
Great to know it, thanks.